cmd/ci-operator/main.go

```diff
@@ -422,17 +423,26 @@ func (o *options) Run() error {
 			return fmt.Errorf("could not initialize namespace: %v", err)
 		}
 
+		client, err := coreclientset.NewForConfig(o.clusterConfig)
+		if err != nil {
+			return fmt.Errorf("could not get core client for cluster config: %v", err)
+		}
+
 		// execute the graph
 		suites, err := steps.Run(ctx, nodes, o.dry)
 		if err := o.writeJUnit(suites, "operator"); err != nil {
 			log.Printf("warning: Unable to write JUnit result: %v", err)
 		}
 		if err != nil {
+			eventRecorder(client, o.namespace).Event(&coreapi.ObjectReference{Namespace: o.namespace}, coreapi.EventTypeNormal, "Steps",
+				fmt.Sprintf("Refs:%v Steps Error:%s", o.jobSpec.Refs, err))
 			return fmt.Errorf("could not run steps: %v", err)
 		}
 
 		for _, step := range postSteps {
 			if err := step.Run(ctx, o.dry); err != nil {
+				eventRecorder(client, o.namespace).Event(&coreapi.ObjectReference{Namespace: o.namespace}, coreapi.EventTypeNormal, "PostSteps",
```

Comment:
I would like the message to look more like the description we put on the namespace (see below)

Reply:
@smarterclayton What output we should approach in case of multiple pulls or refs that have multiple pulls as well?

Reply:
Something like 
``` go
eventRecorder(client, o.namespace).Event(&coreapi.ObjectReference{Namespace: o.namespace}, coreapi.EventTypeNormal, "References",
	fmt.Sprintf("Authors %s in Job: %s with baseSHA: %s\n", func() []string {
		var results []string
		for _, pull := range o.jobSpec.Refs.Pulls {
			out, err := json.Marshal(pull.Author + " with PR: " + strconv.Itoa(pull.Number))
			if err != nil {
				return []string{}
			}
			results = append(results, string(out))
		}
		return results
	}(), o.jobSpec.Job, o.jobSpec.Refs.BaseSHA))
```
With message output will look like:
```
Authors ["droslean with PR: 243" "clayton with PR: 221"] in Job: test-presubmit with baseSHA: 4b75c3617b0f591a8ca32c9ef311750d75b9a3d0
```

Reply:
I would recommend not trying to recalculate the message yourself, but share the description calculation string from below (`jobDescription`)

Reply:
@smarterclayton I created eventJobDescription() for this. The comment below are outdated.